SofaKernel/framework/sofa/core/objectmodel/BaseData.h

```diff
@@ -157,25 +157,25 @@ class SOFA_CORE_API BaseData : public DDGNode
     virtual void releaseAspect(int aspect) = 0;
 
     /// Get a help message that describes this %Data.
-    const char* getHelp() const { return help; }
+    const std::string& getHelp() const { return help; }
 
     /// Set the help message.
-    void setHelp(const char* val) { help = val; }
+    void setHelp(const std::string& val) { help = val; }
 
     /// Get owner class
-    const char* getOwnerClass() const { return ownerClass; }
+    const std::string& getOwnerClass() const { return ownerClass; }
```

Comment:
maybe for backport it would be good to keep some version returing c_char . Where you do the conversion.

Reply:
It would definitely reduce the amount of compile errors to fix in the CI.. but I only delays the moment these things will have to be fixed. I can leave a method with a [[deprecated]] tag maybe..?

Reply:
ok I see. Then replace the string.
I will do the conversion on my side if needed ;)